### PR TITLE
[ADDED] multi-subject filtering (and associated transforms) to stream sources

### PR DESCRIPTION
**Pull request by jnmoyne**
Adds multi-subject filtering (and associated transforms) to stream sources (see https://github.com/nats-io/nats-server/pull/4276)

### CODE REVIEW

**Comment by ripienaar** [comment]
Seems good, but will block till its in the server.

File: api/streams.go [hunk 1]
```diff
@@ -556,13 +556,14 @@ type Placement struct {
...
-	FilterSubject        string          `json:"filter_subject,omitempty"`
-	SubjectTransformDest string          `json:"subject_transform_dest,omitempty"`
+	Name                 string                   `json:"name"`
+	External             *ExternalStream          `json:"external,omitempty"`
+	Lag                  uint64                   `json:"lag"`
+	Active               time.Duration            `json:"active"`
+	Error                *ApiError                `json:"error,omitempty"`
+	FilterSubject        string                   `json:"filter_subject,omitempty"`
+	SubjectTransformDest string                   `json:"subject_transform_dest,omitempty"`
+	SubjectTransforms    []SubjectTransformConfig `json:"subject_transforms,omitempty"`
```
Comment (ripienaar) [inline api/streams.go:566]: The above `SubjectTransformDest` stays? Isn't it redundant? With `StreamSource` not having it is it needed?

**Review by ripienaar — COMMENTED** [review]
just some questions

File: api/streams.go (hunk 1 above)
Comment (jnmoyne) [reply api/streams.go:566]: oh good catch, don't know why StreamSource didn't have it here (in in the server). It's there if you use just one filter/transform. Added it now.

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**